pkg/util.go

```diff
@@ -132,3 +132,16 @@ func GenerateRandomID() int {
 	id := rand.Intn(1000000000) // Adjust the range as needed
 	return id
 }
+
+func MakeCurlCommand(method string, url string, header map[string]string, body string) string {
+	curl := fmt.Sprintf("curl --request %s \\\n", method)
+	curl = curl + fmt.Sprintf("  --url %s \\\n", url)
+	for k, v := range header {
+		if k == "Content-Length" {
```

Comment:
Have you debugged why providing content-length in curl is throwing an error when you use the curl..?

Reply:
@charankamarapu I found the issue. It is happening because of the extra spaces before "url" field of --data.
with extra space             |  without extra space
:-------------------------:|:-------------------------:
![image](https://github.com/keploy/keploy/assets/91385321/20e6f500-39fa-43fd-9d1d-4368b9a5bd2c) |  ![image](https://github.com/keploy/keploy/assets/91385321/49793be2-b138-4ba0-a22d-24bd6af0234c)

Reply:
Ok then try to find out why there is an extra space and remove it. Add content-length header also in the the curl.

Reply:
the extra spaces is because curl command is part of req in .yaml. The only solution is to make curl command a top level field in .yaml. 
when selecting curl command, you can see that there will also be extra spaces :
![image](https://github.com/keploy/keploy/assets/91385321/7b2167af-297a-482a-9f54-92ed4eb22cd4)

Reply:
Oh I see, I guess keeping curl in top level field looks fine for me. Because curl section shouldn't be under spec, as spec majorly tells how keploy decoded its req and resp. Put the curl section before spec or may be at the last.